src/lib/sql/sql_pipeline.cpp

```diff
@@ -73,125 +110,134 @@ const std::vector<std::shared_ptr<AbstractLQPNode>>& SQLPipeline::get_optimized_
+      _failed_pipeline_statement = pipeline;
+
+      // Don't keep bad values
+      _optimized_logical_plans.clear();
+      throw;
     }
-  } catch (const std::exception& exception) {
-    // Don't keep bad values
-    _optimized_logical_plans.clear();
-    throw std::runtime_error("Error while optimizing query plan:\n  " + std::string(exception.what()));
   }
 
   // The optimizer works on the original unoptimized LQP nodes. After optimizing, the unoptimized version is also
-  // optimized, which could lead to subtle bugs. optimized_logical_plans then holds hold the original values now.
+  // optimized, which could lead to subtle bugs. optimized_logical_plan holds the original values now.
   // As the unoptimized LQP is only used for visualization, we can afford to recreate it if necessary.
   _unoptimized_logical_plans.clear();
 
   return _optimized_logical_plans;
 }
 
-const SQLQueryPlan& SQLPipeline::get_query_plan() {
-  if (_query_plan) {
-    return *(_query_plan.get());
+const std::vector<std::shared_ptr<SQLQueryPlan>>& SQLPipeline::get_query_plans() {
+  if (!_query_plans.empty()) {
+    return _query_plans;
   }
 
-  const auto& lqp_roots = get_optimized_logical_plans();
-  auto plan = std::make_unique<SQLQueryPlan>();
+  Assert(!_requires_execution || _pipeline_was_executed,
+         "Cannot compile all statements without executing the previous ones.");
 
-  const auto started = std::chrono::high_resolution_clock::now();
+  _query_plans.reserve(_num_statements);
+  for (auto& pipeline : _sql_pipeline_statements) {
+    try {
+      _query_plans.push_back(pipeline->get_query_plan());
+    } catch (const std::exception& exception) {
+      _failed_pipeline_statement = pipeline;
 
-  try {
-    for (const auto& node : lqp_roots) {
-      auto op = LQPTranslator{}.translate_node(node);
-      plan->add_tree_by_root(op);
+      // Don't keep bad values
+      _query_plans.clear();
+      throw;
     }
-  } catch (const std::exception& exception) {
-    throw std::runtime_error("Error while translating query plan:\n  " + std::string(exception.what()));
-  }
-
-  if (_use_mvcc) {
-    plan->set_transaction_context(_transaction_context);
   }
 
-  const auto done = std::chrono::high_resolution_clock::now();
-  _compile_time_sec = std::chrono::duration_cast<std::chrono::microseconds>(done - started);
-
-  _query_plan = std::move(plan);
-  return *(_query_plan.get());
+  return _query_plans;
 }
 
-const std::vector<TaskSet>& SQLPipeline::get_task_sets() {
-  if (!_op_task_sets.empty()) {
-    return _op_task_sets;
+const std::vector<std::vector<std::shared_ptr<OperatorTask>>>& SQLPipeline::get_tasks() {
+  if (!_tasks.empty()) {
+    return _tasks;
   }
 
-  const auto& query_plan = get_query_plan();
-  _op_task_sets.reserve(query_plan.tree_roots().size());
+  Assert(!_requires_execution || _pipeline_was_executed,
+         "Cannot generate tasks for all statements without executing the previous ones.");
```

Comment:
"One ore more SQL statement is dependent on the execution of a previous one. Call [...] one by one."?